contrib/codegen/src/database.rs

```diff
@@ -56,42 +56,45 @@ fn parse_invocation(attr: TokenStream, input: TokenStream) -> Result<DatabaseInv
     })
 }
 
+#[allow(non_snake_case)]
 pub fn database_attr(attr: TokenStream, input: TokenStream) -> Result<TokenStream> {
     let invocation = parse_invocation(attr, input)?;
 
     // Store everything we're going to need to generate code.
-    let connection_type = &invocation.connection_type;
+    let conn_type = &invocation.connection_type;
     let name = &invocation.db_name;
-    let request_guard_type = &invocation.type_name;
+    let guard_type = &invocation.type_name;
     let vis = &invocation.visibility;
-    let pool_type = Ident::new(&format!("{}Pool", request_guard_type), request_guard_type.span());
+    let pool_type = Ident::new(&format!("{}Pool", guard_type), guard_type.span());
     let fairing_name = format!("'{}' Database Pool", name);
+    let span = conn_type.span().into();
 
     // A few useful paths.
-    let databases = quote!(::rocket_contrib::databases);
-    let r2d2 = quote!(#databases::r2d2);
+    let databases = quote_spanned!(span => ::rocket_contrib::databases);
```

Comment:
Should these have the `span` of the `conn_type` or the default? This looks like the same thing that caused #839.

Reply:
Yeah, I considered that. I _think_ the reason why this is okay is because the entire thing now has that `Span`, as opposed to some parts having it and others not. Perhaps you could add a test, like `core/lib/tests/expansion.rs` to ensure this is actually the case?

Reply:
Ah, I checked again and this looks okay after all. These aren't local variables, they're absolute paths in a use. I'll try to add test tonight or tomorrow to be sure.

Reply:
I think a test is of limited utility after all. After trying to write a macro that expanded to a `#[database(...)]` and encountering "must be a string literal", I can't think of much of a reason to *want* to write a database guard inside a macro. Since these are 1) not local variables and 2) the same span, I think it's fine.